src/rgw/driver/rados/rgw_bucket_sync.cc

```diff
@@ -984,6 +984,20 @@ void RGWBucketSyncPolicyHandler::get_pipes(std::set<rgw_sync_bucket_pipe> *_sour
   }
 }
 
+bool RGWBucketSyncPolicyHandler::bucket_exports_object(const std::string& obj_name, const RGWObjTags* tags) const {
+  if (bucket_exports_data()) {
+    for (auto& entry : target_pipes.pipe_map) {
+      auto& filter = entry.second.params.source.filter;
+      if (filter.check_prefix(obj_name) &&
+	(tags == nullptr || filter.check_tags(tags->get_tags()))) {
+	return true;
```

Comment:
this `tags == nullptr` part seems backwards to me. if a [bucket replication policy](https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketReplication.html) contains a tag-based filter like:
```
<Filter>
    <Tag>
        <Key>string</Key>
        <Value>string</Value>
    </Tag>
</Filter>
```
then we should only sync objects that have a matching tag. if an object has no tags, we shouldn't return true here

it seems like `bucket_exports_object()` should require a `tags` argument (take as a const ref), and callers should pass an empty RGWObjTags instance rather than nullptr

that requires changes to RGWPutObj to avoid passing in nullptr. i prepared https://github.com/ceph/ceph/pull/58291 as an example - if that's useful, feel free to include that commit here